### DISCUSSION TITLE
Is it possible to hide Admonitions from search?

### DISCUSSION THREAD

**Discussion by BrixsterPlanet** [squidfunk/mkdocs-material#5994] [Q&A] [answered]
Hello! I've been trying to use the { data-search-exclude } attribute to hide a remark from the search page preview.
```
!!! warning "Admonition title"
    Content
{ data-search-exclude }
```

While this method successfully hides the content from search, the title still appears:

<img width="328" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/58347007/2544f7ec-efc9-460c-bae3-924c401a37db">

The admonition is not relevant to the core page content (it is created automatically by a Python macro) and therefore should not appear in the search results.

How can I go about fixing this issue?

**Comment by squidfunk** [comment, marked as answer]
I think it might be possible to use [`md_in_html`](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#markdown-in-html) and switch to the following syntax:

``` html
<div data-search-exclude markdown>

!!! warning "Admonition title"
    Content

</div>
```

**Reply by BrixsterPlanet** [reply]
Thanks for the quick response, @squidfunk. I appreciate all the work you do.

With that said, I _do_ wish that this was a feature native to Material for MkDocs (by using the data-search-exclude attribute). It feels unusual that I need to combine two features with an additional plugin and some HTML hacking.

However, that solution worked for me, so I will mark your reply as correct.

With that said, this is a perfectly acceptable solution that I implemented in my macro script successfully.

Thank you!

**Reply by squidfunk** [reply]
It's not a limitation of Material for MkDocs, but of [Python Markdown](https://github.com/Python-Markdown/markdown/). AFAIK, it's currently not possible to attach attributes to an admonition. You can raise this to the maintainers of Python Markdown (maybe there already is such an issue), but we can't do anything about it on our side.
